packages/react/src/globals/scss/_carbon-theme.scss

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright IBM Corp. 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+@import '~carbon-components/scss/globals/scss/vendor/@carbon/elements/scss/themes/theme-maps';
+
+/// @param {Map} $theme-map [$carbon--theme] The theme map to base the result on.
+/// @return {Map} A Carbon theme map whose valus are CSS custom properties.
+@function carbon-get-theme-map-with-custom-properties(
+  $theme-map: $carbon--theme
+) {
+  $theme-map--with-custom-properties: ();
+  @each $key, $value in $theme-map {
+    $theme-map--with-custom-properties: map-merge(
+      $theme-map--with-custom-properties,
+      (
+        $key: var(--#{$key}, $value),
+      )
+    );
+  }
+  @return $theme-map--with-custom-properties;
+}
+
+/// Emits CSS custom properties from a Carbon theme map.
+/// @param {Map} $theme-map [$carbon--theme] The theme map to emit CSS custom properties from.
+/// @example @include carbon--theme--custom-properties($carbon--theme--white);
+@mixin carbon--theme--custom-properties($theme-map: $carbon--theme) {
```

Comment:
Should we move these functions, mixins, maps to `@carbon/themes` so they could get used in other component implementations? If so, can we do that in a future effort or does it need done in this PR?

Reply:
this would depend on what we plan to do with the theme maps right? I think these mixins may not be needed if we export custom properties to begin with

Reply:
True that. I think we're good as you have it because in creating the `@carbon/styles` package, we'll be moving Sass around anyway.